manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.installer.yaml

```
@@ -0,0 +1,17 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.4.0.schema.json
+
+PackageIdentifier: Yandex.Browser
+PackageVersion: 23.1.3.952
+InstallerSuccessCodes:
+- 1
+- 3
+Installers:
+- Architecture: x64
+  InstallerType: exe
+  InstallerUrl: https://cachev2-ams01.cdn.yandex.net/download.cdn.yandex.net/browser/int/23_1_3_949_40687/en/Yandex.exe
+  InstallerSha256: 0159a0da24910b6b9cbdf888f82862e76bdc334d467f0156c279a82fe55eb449

```

```suggestion
  InstallerSha256: 0159a0da24910b6b9cbdf888f82862e76bdc334d467f0156c279a82fe55eb449
  ProductCode: YandexBrowser
```
For better upgrade matching

---

If `productcode` is defined then upgrade will able to find the app to uninstall before installing new version?

---

No, ProductCode just defines the registry key where the installer writes its uninstall entries and related metadata for the package. It is one of the heuristics WinGet uses to find a match for a package.

The behavior you described can be defined by setting `UpgradeBehavior: uninstallPrevious` in the manifest for packages that install side-by-side. Most packages usually upgrade gracefully by just calling the installer.

---

How do you figure `ProductCode` for some specific package?

---

For MSI Installers, you can get it by running this command in Windows PowerShell

`get-wmiobject Win32_Product | Sort-Object Name | Format-Table IdentifyingNumber, Name -AutoSize`

For other installers, you would have to manually search the registry or use some PowerShell scripting to get the uninstall keys. I would recommend using [SandboxTest.ps1](https://github.com/microsoft/winget-pkgs/blob/master/Tools/SandboxTest.ps1) script before submitting a manifest that will display all Registry entries for you.

---

Is there some way to get ProductCode from WinGetSvc-Validation logs when the package gets validated on Azure servers?
Don't have Windows Sandbox on my system :\

---

@Andrej730 In that case, you can resort to using a PowerShell script to retrieve information about the installed softwares on your machine. Some google searches led to me to this article https://powershell.one/code/5.html, the property that you're looking for is `PSChildName`